sdk/keyvault/keyvault-common/src/keyVaultAuthenticationPolicy.ts

```diff
@@ -156,25 +175,90 @@ export function createKeyVaultChallengeCallbacks(
 
     const accessToken = await getAccessToken([scope], {
       ...getTokenOptions,
+      enableCae: true,
       tenantId: parsedChallenge.tenantId,
     });
 
     if (!accessToken) {
-      return false;
+      // No access token provided, treat as no-op
+      return response;
     }
 
     request.headers.set("Authorization", `Bearer ${accessToken.token}`);
 
     challengeState = {
       status: "complete",
       scopes: [scope],
+      tenantId: parsedChallenge.tenantId,
     };
 
-    return true;
+    // We have a token now, so try send the request again
+    return next(request);
+  }
+
+  async function handleCaeChallenge(
+    request: PipelineRequest,
+    response: PipelineResponse,
+    next: SendRequest,
+  ): Promise<PipelineResponse> {
+    // Cannot handle CAE challenge if a regular challenge has not been completed first
+    if (challengeState.status !== "complete") {
+      return response;
+    }
+
+    // If status is not 401, this is a no-op
+    if (response.status !== 401) {
+      return response;
+    }
+
+    const getTokenOptions = requestToOptions(request);
+
+    const challenge = response.headers.get("WWW-Authenticate");
+    if (!challenge) {
+      return response;
+    }
+    const { claims: base64EncodedClaims, error }: WWWAuthenticate =
+      parseWWWAuthenticateHeader(challenge) || {};
```

Comment:
```suggestion
      parseWWWAuthenticateHeader(challenge);
```

I think the signature still returns an object or throws so the `default` may not be needed?